src/Tmds.Ssh/SshAgent.cs

```diff
@@ -0,0 +1,279 @@
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Sockets;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Tmds.Ssh
+{
+    // https://datatracker.ietf.org/doc/html/draft-miller-ssh-agent - SSH Agent Protocol
+    sealed class SshAgent : IDisposable
+    {
+        private const MessageId SSH_AGENTC_REQUEST_IDENTITIES = (MessageId)11;
+        private const MessageId SSH_AGENTC_SIGN_REQUEST = (MessageId)13;
+        private const MessageId SSH_AGENT_FAILURE = (MessageId)5;
+        private const MessageId SSH_AGENT_IDENTITIES_ANSWER = (MessageId)12;
+        private const MessageId SSH_AGENT_SIGN_RESPONSE = (MessageId)14;
+        private const uint SSH_AGENT_RSA_SHA2_256 = 2;
+        private const uint SSH_AGENT_RSA_SHA2_512 = 4;
+
+        private static EndPoint? _defaultEndPoint;
+
+        public struct Identity
+        {
+            public string Comment { init; get; }
+            public byte[] PublicKey { init; get; }
+        }
+
+        public static EndPoint? DefaultEndPoint
+        {
+            get
+            {
+                string? sshAuthSock = Environment.GetEnvironmentVariable("SSH_AUTH_SOCK");
+
+                if (OperatingSystem.IsWindows())
```

Comment:
@jborean93 assuming you have a Windows machine/VM, is this something you'd want to look into?

Reply:
Will try it out and get back to you.

Reply:
On Windows it looks like it doesn't use a socket at all but rather a named pipe called `openssh-ssh-agent`. You can use [NamedPipeClientStream](https://learn.microsoft.com/en-us/dotnet/api/system.io.pipes.namedpipeclientstream?view=net-9.0) to connect to it

```csharp
using var agentPipe = new NamedPipeClientStream("openssh-ssh-agent");
await agentPipe.ConnectAsync();
...
```

There's probably some considerations needed around

+ Whether any custom [PipeOption](https://learn.microsoft.com/en-us/dotnet/api/system.io.pipes.pipeoptions?view=net-9.0) is needed, like `Asynchronous`
+ Whether we should specify [TokenImpersonationLevel.Anonymous](https://learn.microsoft.com/en-us/dotnet/api/system.security.principal.tokenimpersonationlevel?view=net-9.0) or `Identification` to stop the service from pretending to be us
  + I'm not aware of any reason why it would need to do so but it's worth looking into

The pipe implements a `Stream` and not a `Socket` which will have some work needed to fit into the existing API model.

Reply:
> The pipe implements a Stream and not a Socket which will have some work needed to fit into the existing API model.

For this, you can change `SocketSshConnection` to accept a `Stream` instead of `Socket` and for the `Socket`-case create a `NetworkStream`. That changes will be useful for proxy jumping. Feel free to rename the class (e.g. `StreamSshConnection`.